### PR TITLE
[FEA-38] Migrate www.ably.com to ably.com

### PR DESCRIPTION
**Pull request by twist900**
ref: [FEA-38](https://ably.atlassian.net/browse/FEA-38)

### CODE REVIEW

**Review by tbedford — APPROVED** [review]

**Comment by tbedford** [comment]
Please note the review app failed because `.code` files were modified and JSBin hashes were not updated:

```
Captain! We’ve been hit!
JsBinClient::AuthenticationRequired: The API key you provided is not valid
```

I think data/jsbins.yaml also needs updating.

**Comment by twist900** [comment]
> Please note the review app failed because `.code` files were modified and JSBin hashes were not updated:
> 
> ```
> Captain! We’ve been hit!
> JsBinClient::AuthenticationRequired: The API key you provided is not valid
> ```
> 
> I think `data/jsbins.yaml` also needs updating.

Thank you @tbedford, is this `JsBinClient::AuthenticationRequired: The API key you provided is not valid` related to [WEB-1406](https://ably.atlassian.net/browse/WEB-1406)?

I also get this when trying to generate the new jsbin hash (`bundle exec nanoc compile`)

**Comment by twist900** [comment]
ah, RTFM! I went through the [jsbin readme section](https://github.com/ably/docs/blob/a842a2fe1eccc3399c49827f1f386bf18381ce19/README.md#L48) more carefully now and set the api key locally -- all is generated successfully now 👍

File: content/client-lib-development-guide/versions/features-0-8__0-9.diff [hunk 1]
```diff
@@ -31,16 +31,16 @@ index bff9943..5f87d5d 100644
 -As a result, this document outlines the complete feature set of both the REST and Realtime client libraries.  It is expected that every client library developer refers to this document to ensure that their client library provides the same API and features as the existing Ably client libraries. In addition to this, it is essential that there is test coverage over all of the features described below.
```
Comment (kennethkalmer) [inline content/client-lib-development-guide/versions/features-0-8__0-9.diff:31]: What is the difference here, this is a confusing change... Whitespace maybe? Character encoding?

File: content/tutorials/tfl-live-tracking-fitbit.textile [hunk 2]
```diff
@@ -183,7 +183,7 @@ h2(#walking-time). Step 5 - Get location & find walking time
 Now we will get the location and update the walking time to the user selected station.
 So for that we will need to use the Google API to get the walking time.
 Follow the steps "here":https://developers.google.com/maps/documentation/distance-matrix/get-api-key, to get the API key.
-You can then subscribe to the "TFL Tube Schedule":https://www.ably.io/hub/products/10 via the "Ably Hub":https://www.ably.io/hub
+You can then subscribe to the "TFL Tube Schedule":https://ably.com/hub/products/10 via the "Ably Hub":https://ably.com/hub
```
Comment (kennethkalmer) [inline content/tutorials/tfl-live-tracking-fitbit.textile:186]: https://ably.com/hub/ably-tfl/tube

File: content/tutorials/live-bitcoin-pricing.textile [hunk 3]
```diff
@@ -211,9 +211,9 @@ We started with initializing the "Ably realtime client library":/realtime using
...
+After initializing the Ably realtime client library, we'll need to attach it to the channel on which the required data is being published and then subscribe to it to receive updates as they happen continuously. All the data sharing in Ably happens over fundamental units called "channels":https://ably.com/channels. The @ably.channels.get()@ lets you implicitly attach to a channel. We know the name of the channel from the "product documentation on the Ably Hub":https://ably.com/hub/products/20#documentation. In this case, the channel name to be used is @[product:ably-coindesk/bitcoin]bitcoin:usd@.
```
Comment (kennethkalmer) [inline content/tutorials/live-bitcoin-pricing.textile:214]: https://ably.com/hub/ably-coindesk/bitcoin

File: content/tutorials/web-rtc-data-channels.textile [hunk 4]
```diff
@@ -464,7 +464,7 @@ To try this example yourself, <a href="https://datachannels.herokuapp.com/" id="
 
 h2(#conclusion). Conclusion
 
-In this tutorial, we have seen how to use WebRTC data channels in conjunction with Ably to create realtime chats. Do you have any thoughts on making this lesson better? "Let us know":mailto:devrel@ably.io.
+In this tutorial, we have seen how to use WebRTC data channels in conjunction with Ably to create realtime chats. Do you have any thoughts on making this lesson better? "Let us know":mailto:devrel@ably.com.
```
Comment (kennethkalmer) [inline content/tutorials/web-rtc-data-channels.textile:467]: `developer.relations@ably.com`

File: content/tutorials/web-rtc-video-calling.textile [hunk 5]
```diff
@@ -450,7 +450,7 @@ Please note that if you are using a single Pc to test the demo, please use the s
 
 h2(#conclusion). Conclusion
 
-In this tutorial, we have seen how to use WebRTC in conjunction with Ably to create real time video call app. Have any thoughts on how we could make this lesson better? "Let us know":mailto:devrel@ably.io.
+In this tutorial, we have seen how to use WebRTC in conjunction with Ably to create real time video call app. Have any thoughts on how we could make this lesson better? "Let us know":mailto:devrel@ably.com.
```
Comment (kennethkalmer) [inline content/tutorials/web-rtc-video-calling.textile:453]: `developer.relations@ably.com`

**Review by kennethkalmer — CHANGES_REQUESTED** [review]
Few small changes I picked up, including email addresses that don't map 1:1 between the domains. @tbedford in the `diff` files there are some strange changes, do you know if this is because of a character encoding issue?

**Comment by twist900** [comment]
@kennethkalmer all streamer product links are now renamed throughout the docs repo, thanks 👍 

I also found a broken link/missing product, please see 8519cbb

regenerated jsbin hashes just to be sure ✅

File: content/client-lib-development-guide/versions/features-0-8__0-9.diff (hunk 1 above)
Comment (tbedford) [reply content/client-lib-development-guide/versions/features-0-8__0-9.diff:31]: I'll have a look later. I'm not sure what the .diff files are or how they are generated.

**Comment by tbedford** [comment]
> Thank you @tbedford, is this `JsBinClient::AuthenticationRequired: The API key you provided is not valid` related to [WEB-1406](https://ably.atlassian.net/browse/WEB-1406)?

It is related to https://ably.atlassian.net/browse/WEB-1407 for the review apps not building.

File: content/client-lib-development-guide/versions/features-0-8__0-9.diff (hunk 1 above)
Comment (tbedford) [reply content/client-lib-development-guide/versions/features-0-8__0-9.diff:31]: I think these `.diff` files are created by `nanoc` when it does the doc versioning. I don't think we should be modifying these files should we, as I think they are machine generated? To be honest I'm not sure about these files. Would @fliptopbox know?

Comment (kennethkalmer) [reply content/client-lib-development-guide/versions/features-0-8__0-9.diff:31]: @tomczoink could you help us here with a quick look at what is happening on this line, and also where the `.diff` files come from?

Comment (tomczoink) [reply content/client-lib-development-guide/versions/features-0-8__0-9.diff:31]: This version was a bit before me unfortunately, but I think it's partially hand-edited but not 100% sure.

**Comment by twist900** [comment]
@kennethkalmer @tbedford @tomczoink I did not look at how these doc versions are generated, but these are the versions that are available from the drop down selection, so if we don't manually update previously created doc versions, the links on those pages will be broken.

I ran locally the changes, all pages and their versions work properly and include correct links. Merge?

**Comment by tbedford** [comment]
@twist900 - yes we should update all versions of the docs - it was just the `.diff` files - I thought they were machine generated and so should not be modified, but I don't totally understand where they come from.

File: content/api-streamer/examples.textile [hunk 6]
```diff
@@ -97,25 +97,6 @@ h3. Live Demo:
   </tr>
 </table>
 
-h2(#hackernews). Hacker News - a live stream of the latest tech and startup links submitted on HN
```
Comment (mattheworiordan) [inline content/api-streamer/examples.textile:100]: Why is this deleted OOI?

File: content/client-lib-development-guide/versions/features-0-8__0-9.diff (hunk 1 above)
Comment (mattheworiordan) [reply content/client-lib-development-guide/versions/features-0-8__0-9.diff:31]: Yup, don't change diff files, they are auto-generated and this will create confusion as this is a record of what changed at that time, so we can't change it after that.

**Review by mattheworiordan — CHANGES_REQUESTED** [review]
Wowza, that's a lot of changes 👍 

One comment on diffs, otherwise 👍

File: content/api-streamer/examples.textile (hunk 6 above)
Comment (kennethkalmer) [reply content/api-streamer/examples.textile:100]: @mattheworiordan the hub product at https://www.ably.io/hub/products/15 no longer exists

File: content/client-lib-development-guide/versions/features-0-8__0-9.diff (hunk 1 above)
Comment (kennethkalmer) [reply content/client-lib-development-guide/versions/features-0-8__0-9.diff]: I removed the diffs from this change

**Review by kennethkalmer — APPROVED** [review]
I removed the `.diff` files from the big commit just like @mattheworiordan asked. This looks good to me, if we get the tick from @mattheworiordan & @tbedford then this can get merged and onto the website as well.

Thanks again @twist900 for this enormous effort!

**Review by tbedford — APPROVED** [review]
